### DISCUSSION TITLE
Schema extension: where is ory.sh/kratos property documented?

### DISCUSSION THREAD

**Discussion by space88man** [ory/kratos#3389] [Q&A] [answered]
In kratos identities there is an identity schema extension property `ory.sh/kratos` used to indicate credential identifiers, email recovery etc.

Where is this documented (i.e. what is the schema for this property)?

```
"ory.sh/kratos": {
            "credentials": {
              "password": {
                "identifier": true
              },
              "webauthn": {
                "identifier": true
              },
              "totp": {
                "account_name": true
              }
            },
            "recovery": {
              "via": "email"
            },
            "verification": {
              "via": "email"
            }
          },
```

**Comment by vinckr** [comment, marked as answer]
Hello @space88man 

Do you mean JSON schema: http://json-schema.org/draft-07/schema ? 

Generally the identity schema is really flexible, can you elaborate a bit more on your use case maybe? 

The ory.sh/kratos property in the identity schema is documented in the Ory Kratos documentation, specifically in the sections about customizing the identity schema.

Briefly explaining each object: 

`credentials`: This object is used to specify which fields should be used as identifiers for different authentication methods. For example, in the provided code, both the password and webauthn methods use the current field (where this ory.sh/kratos object is defined) as an identifier. This means that the user can use the value of this field to log in with either a password or WebAuthn.
`totp`: This stands for Time-based One-Time Password, a common method for two-factor authentication. The account_name property being set to true indicates that the current field should be used as the account name for TOTP.
`recovery`: This object specifies how the user can recover their account. The via property being set to email means that the user can recover their account using their email.
`verification`: Similar to recovery, this object specifies how the user's account can be verified. Again, via being set to email means that the user's email is used for verification.

Here are the relevant documentation links where these properties are discussed:

- [Customize identity schemas](https://www.ory.sh/docs/kratos/manage-identities/customize-identity-schema)
- [Account recovery and password reset](https://www.ory.sh/docs/kratos/self-service/flows/account-recovery-password-reset)
- [WebAuthn Configuration](https://www.ory.sh/docs/kratos/passwordless/webauthn)
- [Multiple identifiers](https://www.ory.sh/docs/kratos/manage-identities/customize-identity-schema)
- [Password login identifier](https://www.ory.sh/docs/kratos/manage-identities/customize-identity-schema)
- [WebAuthn and FIDO2](https://www.ory.sh/docs/kratos/mfa/webauthn-fido-yubikey)

**Reply by space88man** [reply]
I was hoping that there was a schema for the object of the `ory.sh/kratos` property rather than definition-by-example.
